packages/create-amplify/src/get_project_root.ts

```diff
@@ -37,7 +37,17 @@ export const getProjectRoot = async () => {
       LogLevel.DEBUG
     );
     printer.log(`Creating directory ${projectRoot}`, LogLevel.DEBUG);
-    await fsp.mkdir(projectRoot, { recursive: true });
+    try {
+      await fsp.mkdir(projectRoot, { recursive: true });
+    } catch (err) {
+      if (path.isAbsolute(projectRoot)) {
```

Comment:
In what conditions would a `projectRoot` not be an absolute path? 

https://github.com/aws-amplify/amplify-backend/pull/1071/files#diff-17a8da21bcc08aa19a0287db1113f604841ad4119d227505260fdc18ac081181R26-R28

Reply:
I can't think of a case at the moment where projectRoot not be an absolute path, but I put it in explicitly.

Is it necessary?

Reply:
Thank you feedback !!! @edwardfoyle @Amplifiyer 

I try to do this task!
- [ ] remove condition
- [ ] add `AmplifyError`

Reply:
@Amplifiyer @edwardfoyle 

I came up with three error pattern.
should select plan3 right?

```
// plan1: use printer
try {
    await fsp.mkdir(projectRoot, {recursive: true});
} catch (err) {
    printer.log(
        `Failed to create directory at ${projectRoot}. Ensure this is the correct path and you have write permissions to this location.`,
        LogLevel.ERROR
    );
    throw err;
}
------------------
// plan2: direct throw new Error
try {
    await fsp.mkdir(projectRoot, {recursive: true});
} catch (err) {
    throw new Error(`Failed to create directory at ${projectRoot}. Ensure this is the correct path and you have write permissions to this location.`);
}
------------------
// plan3: use AmplifyUserError
import { AmplifyUserError } from '@aws-amplify/platform-core';

throw new AmplifyUserError('MultipleSingletonResourcesError', {
    message:
        `Failed to create directory at ${projectRoot}. Ensure this is the correct path and you have write permissions to this location.`,
    resolution: 'Change relative path or Absolute path under root',
});
```

Reply:
Yes plan3 is correct. Make the error name something like `ProjectDirectoryCreateError` and change the message and resolution to something like 

```ts
try {
    await fsp.mkdir(projectRoot, {recursive: true});
} catch (err) {
    throw new AmplifyUserError('MultipleSingletonResourcesError', {
        message:
            `Failed to create project directory`,
        resolution: 'Ensure that ${projectRoot} is the correct path and you have write permissions to this location.',
    });
}
```

Reply:
https://github.com/aws-amplify/amplify-backend/pull/1071/commits/4e070def49e77f1560468835a90fbc837cfe1358